contrib/cli/great_expectations_contrib/commands.py

```diff
@@ -0,0 +1,135 @@
+    These include code style, testing, docstrings, and more.
+    """
+    perform_check(suppress_output=False)
+
+
+def publish_cmd() -> None:
+    """
+    Performs same checks as `check_cmd`; if they pass, the user is prompted to
+    supply PyPi credentials. Valid inputs will result in an uploaded package.
+    """
+    success = perform_check(suppress_output=True)
+    if not success:
+        echo(
+            "Please run the `check` command to diagnose before publishing",
+            "red",
+            bold=True,
+        )
+        return
+
+    echo("All checks have succeeded; you are ready to publish!", "green", bold=True)
+    publish_to_pypi()
+
+
+def perform_check(suppress_output: bool) -> bool:
+    commands = [
+        Command(
+            "black",
+            "black --check .",
+            "Please ensure that your files are linted properly with `black .`",
+        ),
+        Command(
+            "isort",
+            "isort --profile black --check .",
+            "Please ensure that your imports are sorted properly with `isort --profile black .`",
+        ),
+        Command(
+            "pytest",
+            "pytest .",
+            "Please ensure that you've written tests and that they all pass",
+        ),
+        Command(
+            "mypy",
+            "mypy --ignore-missing-imports --disallow-untyped-defs --show-error-codes --exclude venv .",
+            "Please ensure that all functions are type hinted",
+        ),
+    ]
+
+    successes = 0
+    for i, command in enumerate(commands):
+        if run_command(i + 1, command, suppress_output=suppress_output):
+            successes += 1
+
+    is_successful = successes == len(commands)
+    color = "green" if is_successful else "red"
+    echo(
+        f"Summary: [{successes}/{len(commands)}] checks have passed!", color, bold=True
+    )
+
+    return is_successful
+
+
+def publish_to_pypi() -> None:
+    commands = [
+        Command(
+            "wheel",
+            "python setup.py sdist bdist_wheel",
+            "Something went wrong when creating a wheel",
+        ),
+        Command(
+            "twine",
+            "twine upload --repository testpypi dist/*",
+            "Something went wrong when uploading with twine",
+        ),
+    ]
+
+    for i, command in enumerate(commands):
+        if not run_command(i + 1, command):
+            return
+
+    echo(
+        "Successfully uploaded package to PyPi! Congratulations on a job well done :)",
+        "green",
+        bold=True,
+    )
+
+
+def run_command(idx: int, command: Command, suppress_output: bool = False) -> bool:
+    # If suppressed, set STDOUT to dev/null
+    stdout = sys.stdout
+    if suppress_output:
+        sys.stdout = open(os.devnull, "w")
+
+    name, args, err = command
+    echo(f"{idx}) {name}:", "blue", bold=True)
+    result = subprocess.run(
+        args.split(" "), shell=False, stdout=sys.stdout, stderr=sys.stdout
```

Comment:
`[name] + args.split(" ")` if you end up making the 2nd item of `Command` instances (args) not include the name

Reply:
based on the response above (`name` isn't always the first value) I think we're okay with this. I'll rename the vars to make it a bit clearer!